pubsublite-kafka-shim/src/main/java/com/google/cloud/pubsublite/kafka/KafkaExceptionUtils.java

```diff
@@ -0,0 +1,92 @@
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsublite.kafka;
+
+import com.google.cloud.pubsublite.internal.ExtractStatus;
+import io.grpc.StatusException;
+import java.util.concurrent.TimeoutException;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.errors.AuthenticationException;
+import org.apache.kafka.common.errors.AuthorizationException;
+import org.apache.kafka.common.errors.BrokerNotAvailableException;
+import org.apache.kafka.common.errors.InterruptException;
+import org.apache.kafka.common.errors.InvalidRequestException;
+import org.apache.kafka.common.errors.TopicExistsException;
+
+class KafkaExceptionUtils {
+  private KafkaExceptionUtils() {}
+
+  static KafkaException toKafkaException(StatusException source) {
+    switch (source.getStatus().getCode()) {
+      case OK:
+        throw source.getStatus().asRuntimeException();
+      case ABORTED:
+        return new BrokerNotAvailableException("Aborted.", source);
+      case ALREADY_EXISTS:
+        return new TopicExistsException("", source);
+      case CANCELLED:
+        return new BrokerNotAvailableException("Cancelled.", source);
+      case DATA_LOSS:
+        return new KafkaException("Data loss.", source);
+      case DEADLINE_EXCEEDED:
+        return new BrokerNotAvailableException("Deadline exceeded.", source);
+      case FAILED_PRECONDITION:
+        return new InvalidRequestException("Failed precondition.", source);
+      case INTERNAL:
+        return new BrokerNotAvailableException("Internal.", source);
+      case INVALID_ARGUMENT:
+        return new InvalidRequestException("Invalid argument.", source);
+      case NOT_FOUND:
+        return new KafkaException("Not found.", source);
+      case OUT_OF_RANGE:
+        return new KafkaException("Out of range.", source);
+      case PERMISSION_DENIED:
+        return new AuthorizationException("Permission denied.", source);
+      case RESOURCE_EXHAUSTED:
+        return new KafkaException("Resource exhausted.", source);
+      case UNAUTHENTICATED:
+        return new AuthenticationException("Unauthenticated.", source);
+      case UNAVAILABLE:
+        return new BrokerNotAvailableException("Unavailable.", source);
+      case UNIMPLEMENTED:
+        return new KafkaException("Unimplemented.", source);
+      case UNKNOWN:
+        return new KafkaException("Unknown.", source);
+    }
+    return new KafkaException("No case.", source);
+  }
+
+  /**
+   * Transform an exception into a kind that is likely to be thrown through kafka interfaces.
+   *
+   * @param t A throwable to transform.
+   * @return The transformed exception suitable for re-throwing.
+   */
+  static RuntimeException toKafka(Throwable t) {
+    try {
+      throw t;
+    } catch (KafkaException | UnsupportedOperationException | IllegalStateException e) {
+      return e;
+    } catch (InterruptedException e) {
+      return new InterruptException(e);
```

Comment:
Nit; just "return e;"? What's the difference?

Reply:
InterruptException derives from KafkaException and is unchecked- InterruptedException is the java stdlib one thrown when a thread is interrupted.